File: app/routers/themes.py
from fastapi import APIRouter
import json
from ..config import settings
from ..models import ThemeInfo, ThemesResponse

router = APIRouter(prefix="/api", tags=["themes"])


@router.get("/themes", response_model=ThemesResponse)
async def list_themes():
    """List all available poster themes."""
    themes_dir = settings.maptoposter_dir / "themes"
    themes = []

    for theme_file in sorted(themes_dir.glob("*.json")):
        with open(theme_file) as f:
            data = json.load(f)
            themes.append(
                ThemeInfo(
                    id=theme_file.stem,
                    name=data.get("name", theme_file.stem),
                    description=data.get("description"),
                    bg=data.get("bg", "#FFFFFF"),
                    text=data.get("text", "#000000"),
                )
            )

    return ThemesResponse(themes=themes)

File: server/src/routes/themes.js
import { Router } from 'express';
import { readdirSync, readFileSync } from 'fs';
import { join } from 'path';
import { config } from '../config.js';

export const themesRouter = Router();

/**
 * GET /api/themes
 * List all available poster themes.
 */
themesRouter.get('/themes', (req, res) => {
  try {
    const themesDir = config.themesDir;
    const themeFiles = readdirSync(themesDir)
      .filter(file => file.endsWith('.json'))
      .sort();

    const themes = themeFiles.map(file => {
      const filePath = join(themesDir, file);
      const data = JSON.parse(readFileSync(filePath, 'utf-8'));
      const id = file.replace('.json', '');

      return {
        id,
        name: data.name || id,
        description: data.description || null,
        bg: data.bg || '#FFFFFF',
        text: data.text || '#000000',
      };
    });

    res.json({ themes });
  } catch (error) {
    console.error('Failed to load themes:', error);
    res.status(500).json({ error: 'Failed to load themes' });
  }
});
